src/trace_processor/perfetto_sql/stdlib/android/kernel_wakelocks.sql

```diff
@@ -15,33 +15,42 @@
 
 INCLUDE PERFETTO MODULE android.suspend;
 
+INCLUDE PERFETTO MODULE counters.intervals;
+
+CREATE PERFETTO TABLE _kernel_wakelock_track AS
+SELECT
+  id,
+  name,
+  extract_arg(dimension_arg_set_id, 'wakelock_type') AS type
+FROM track AS t
+WHERE
+  type = 'android_kernel_wakelock';
+
 CREATE PERFETTO TABLE _android_kernel_wakelocks_base AS
 WITH
-  raw AS (
+  kernel_wakelock_counter AS (
     SELECT
-      -- Move back by one period here since our ts represents wakelock counts up
-      -- to that point, but counters project forward from their ts.
-      lag(ts) OVER (PARTITION BY track_id ORDER BY ts) AS ts,
-      ts AS next_ts,
-      name,
-      extract_arg(dimension_arg_set_id, 'wakelock_type') AS type,
-      value,
-      lag(value) OVER (PARTITION BY track_id ORDER BY ts) AS lag_value
-    FROM track AS t
-    JOIN counter AS c
-      ON t.id = c.track_id
-    WHERE
-      t.type = 'android_kernel_wakelock'
+      *
+    FROM counter_leading_intervals!((
+        select 
+          c.id,
+          c.ts,
+          c.track_id,
+          c.value
+        from counter c join _kernel_wakelock_track t on t.id = c.track_id
```

Comment:
Writing this as a `WHERE c.track_id IN (select id from _kernel_wakelock_track)` will be more efficient.